### PR TITLE
feat: search product by name

### PR DESCRIPTION
**Pull request by natehishon**

### CODE REVIEW

File: .idea/artifacts/WebServicesProject2_war_exploded.xml [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
   <artifact type="exploded-war" name="WebServicesProject2:war exploded">
-    <output-path>$PROJECT_DIR$/target/WebServicesProject-1.0-SNAPSHOT</output-path>
-    <properties id="maven-jee-properties">
-      <options>
-        <exploded>true</exploded>
-        <module>WebServicesProject2</module>
-        <packaging>war</packaging>
-      </options>
-    </properties>
+    <output-path>$PROJECT_DIR$/out/artifacts/WebServicesProject2_war_exploded</output-path>
```
Comment (RobusK) [inline .idea/artifacts/WebServicesProject2_war_exploded.xml]: whyyyyyyyyyyyyyyyyyyyyyyyyyyyy

File: .idea/artifacts/WebServicesProject2_war.xml [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
   <artifact type="war" name="WebServicesProject2:war">
-    <output-path>$PROJECT_DIR$/target</output-path>
-    <properties id="maven-jee-properties">
-      <options>
-        <module>WebServicesProject2</module>
-        <packaging>war</packaging>
-      </options>
-    </properties>
-    <root id="archive" name="WebServicesProject-1.0-SNAPSHOT.war">
+    <output-path>$PROJECT_DIR$/out/artifacts/WebServicesProject2_war</output-path>
```
Comment (RobusK) [inline .idea/artifacts/WebServicesProject2_war.xml]: whyyyyyyyyyyyyyyyyyyy

File: .idea/misc.xml [hunk 3]
```diff
@@ -10,7 +10,7 @@
       </list>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_10" default="true" project-jdk-name="10" project-jdk-type="JavaSDK">
```
Comment (RobusK) [inline .idea/misc.xml]: aint no jdk10

File: src/main/java/com/webservices/controllers/ProductsController.java [hunk 4]
```diff
@@ -87,4 +87,19 @@ public ProductsController(ProductService productService) {
             return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
         }
     }
+
+
+    @GetMapping("/find-by-name")
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/controllers/ProductsController.java]: make it the second method in every layer

File: src/main/java/com/webservices/controllers/ProductsController.java [hunk 5]
```diff
@@ -87,4 +87,19 @@ public ProductsController(ProductService productService) {
...
+    @ResponseBody
+    public ResponseEntity<?> findProductsByName(@RequestParam String name) {
+        try {
+            List products = productService.findProductsByName(name);
+            if (products == null) {
+                return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            }
+            return new ResponseEntity<>(products, HttpStatus.OK);
+        } catch (Exception e) {
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/controllers/ProductsController.java]: add logging here

File: src/main/java/com/webservices/repositories/ProductDAOImpl.java [hunk 6]
```diff
@@ -37,6 +39,26 @@ public Product getProduct(Integer id) {
...
+        try {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Product> criteriaQuery = criteriaBuilder.createQuery(Product.class);
+            Root<Product> root = criteriaQuery.from(Product.class);
+            criteriaQuery
+                    .select(root)
+                    .where(criteriaBuilder.like(criteriaBuilder.upper(root.get("name")), "%" + name.toUpperCase() + "%"));
+            products = session.createQuery(criteriaQuery).getResultList();
+        } catch (Exception e) {
+            logger.error("Unable to find products by name", e);
```
Comment (mDemianchuk) [inline src/main/java/com/webservices/repositories/ProductDAOImpl.java]: `...by name " + name, e);`

File: .idea/misc.xml (hunk 3 above)
Comment (mDemianchuk) [reply .idea/misc.xml]: just add this file to `.gitignore`

File: .idea/artifacts/WebServicesProject2_war.xml (hunk 2 above)
Comment (mDemianchuk) [reply .idea/artifacts/WebServicesProject2_war.xml]: use output path from master for both

**Pull request merged**